### PR TITLE
Land-1253 chat search doesn't work

### PR DESCRIPTION
**Pull request by mrozanski**
- Ensure decToUd() receives a string type arg
- Default to '0' (decToUd returns empty string when a '0' is passed)
- Avoid running query if search term is empty

before:
Screenshot 2023-11-27 at 7 34 53 PM

### CODE REVIEW

File: ui/src/app.tsx [hunk 1]
```diff
@@ -217,10 +218,7 @@ function ChatRoutes({ state, location, isMobile, isSmall }: RoutesProps) {
               />
             )}
             {isMobile && (
-              <Route
-                path=":ship/search/:query?"
-                element={<MobileChatSearch />}
-              />
+              <Route path=":ship/search/:query?" element={<MobileDmSearch />} />
```
Comment (arthyn) [inline ui/src/app.tsx:221]: Also needs to be changed in `GroupRoutes`

**Review by arthyn — CHANGES_REQUESTED** [review]
LGTM, but needs the same change for Groups app routes

**Review by arthyn — APPROVED** [review]
lgtm

**Review by latter-bolden — APPROVED** [review]
Working for me on chat and DMs for both mobile and desktop

**Pull request merged**